### PR TITLE
feat(autofix): Optional unit test in root cause

### PR DESCRIPTION
**Pull request by jennmueng**
Renders the unit test to reproduce a given root cause. As this unit test is optional, this can be merged without the seer PR.

1. Adds a unit test dropdown w/ description
2. Adds icons and spacing to the root cause context area to make it feel less cluttered and more welcoming.

CleanShot 2024-10-01 at 14 59 26@2x

### CODE REVIEW

File: static/app/components/events/autofix/autofixRootCause.tsx [hunk 1]
```diff
@@ -165,15 +166,37 @@ export function replaceHeadersWithBold(markdown: string) {
...
+    ? getPrismLanguage(unitTestFileExtension)
+    : undefined;
+
+  return (
+    <RootCauseContextContainer>
       {cause.reproduction && (
         <Fragment>
-          <ExpandableInsightContext title={'How to reproduce'}>
+          <ExpandableInsightContext
+            icon={<IconLaptop size="sm" color="subText" />}
```
Comment (roaga) [inline static/app/components/events/autofix/autofixRootCause.tsx:196]: Why a laptop? I don't have a suggestion but maybe there's a better icon for this

File: static/app/components/events/autofix/autofixRootCause.tsx [hunk 2]
```diff
@@ -165,15 +166,37 @@ export function replaceHeadersWithBold(markdown: string) {
...
+  const unitTestFileExtension = cause.unit_test?.file_path
+    ? getFileExtension(cause.unit_test.file_path)
+    : undefined;
+  const unitTestLanguage = unitTestFileExtension
+    ? getPrismLanguage(unitTestFileExtension)
+    : undefined;
+
+  return (
+    <RootCauseContextContainer>
       {cause.reproduction && (
```
Comment (roaga) [inline static/app/components/events/autofix/autofixRootCause.tsx:193]: you have `cause.reproduction` here, but didn't you rename that field in the backend?

File: static/app/components/events/autofix/autofixRootCause.tsx [hunk 3]
```diff
@@ -165,15 +166,37 @@ export function replaceHeadersWithBold(markdown: string) {
...
         <Fragment>
-          <ExpandableInsightContext title={'How to reproduce'}>
+          <ExpandableInsightContext
+            icon={<IconLaptop size="sm" color="subText" />}
+            title={'How to reproduce'}
+            rounded
+          >
             <CauseDescription
               dangerouslySetInnerHTML={{
                 __html: marked(replaceHeadersWithBold(cause.reproduction)),
```
Comment (roaga) [inline static/app/components/events/autofix/autofixRootCause.tsx:202]: Same here. `cause.reproduction` or `cause.reproduction_instructions`?

File: static/app/components/events/autofix/autofixRootCause.tsx [hunk 4]
```diff
@@ -182,7 +205,35 @@ function RootCauseDescription({cause}: {cause: AutofixRootCauseData}) {
...
       )}
-    </Fragment>
+      {cause.unit_test && (
+        <Fragment>
+          <ExpandableInsightContext
+            icon={<IconLab size="sm" color="subText" />}
+            title={'Unit test'}
+            rounded
+          >
+            <CauseDescription
```
Comment (roaga) [inline static/app/components/events/autofix/autofixRootCause.tsx:215]: Should be integrated with reproduction section as discussed

File: static/app/components/events/autofix/types.ts [hunk 5]
```diff
@@ -166,12 +166,19 @@ export type AutofixRootCauseCodeContext = {
...
+  file_path: string;
+  snippet: string;
+};
+
 export type AutofixRootCauseData = {
   code_context: AutofixRootCauseCodeContext[];
   description: string;
   id: string;
   title: string;
   reproduction?: string;
```
Comment (roaga) [inline static/app/components/events/autofix/types.ts:180]: again, reproduction or reproduction_instructions?

File: static/app/components/events/autofix/autofixRootCause.tsx (hunk 2 above)
Comment (jennmueng) [reply static/app/components/events/autofix/autofixRootCause.tsx:193]: `reproduction_instructions` is only for the pydantic model sent to the llm to prompt it

File: static/app/components/events/autofix/autofixRootCause.tsx (hunk 1 above)
Comment (jennmueng) [reply static/app/components/events/autofix/autofixRootCause.tsx:196]: There's a long line of icons that somewhat make sense, maybe the info icon instead?
CleanShot 2024-10-01 at 14 53 18@2x
CleanShot 2024-10-01 at 14 53 05@2x
CleanShot 2024-10-01 at 14 52 54@2x
CleanShot 2024-10-01 at 14 52 46@2x
CleanShot 2024-10-01 at 14 52 43@2x
CleanShot 2024-10-01 at 14 52 31@2x
CleanShot 2024-10-01 at 14 52 14@2x
CleanShot 2024-10-01 at 14 52 08@2x

Comment (roaga) [reply static/app/components/events/autofix/autofixRootCause.tsx:196]: 🤷 Maybe ask Vu, or just use the laptop as a placeholder for now until we can get his input

Comment (jennmueng) [reply static/app/components/events/autofix/autofixRootCause.tsx]: We'll just do info for now and ill follow up for input

**Review by roaga — APPROVED** [review]
Looks good!

**Pull request merged**